main.go

```diff
@@ -16,91 +16,111 @@ package main
 
 import (
 	"context"
-	"flag"
 	"fmt"
 	"os"
-	"strings"
 
 	"github.com/mendersoftware/go-lib-micro/log"
-	"github.com/spf13/viper"
+	"github.com/urfave/cli"
+	"gopkg.in/mgo.v2"
 
 	"github.com/mendersoftware/deployments/config"
 )
 
 func main() {
+	doMain(os.Args)
+}
+
+func doMain(args []string) {
 
 	var configPath string
-	var printVersion bool
-	flag.StringVar(&configPath, "config", "", "Configuration file path. Supports JSON, TOML, YAML and HCL formatted configs.")
-	flag.BoolVar(&printVersion, "version", false, "Show version")
 
-	flag.Parse()
+	app := cli.NewApp()
+	app.Usage = "Deployments Service"
+	app.Version = CreateVersionString()
 
-	if printVersion {
-		fmt.Println("Version:", CreateVersionString())
-		fmt.Println("BuildNumber:", BuildNumber)
-		os.Exit(0)
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:        "config",
+			Usage:       "Configuration `FILE`. Supports JSON, TOML, YAML and HCL formatted configs.",
+			Destination: &configPath,
+		},
 	}
 
-	l := log.New(log.Ctx{})
-
-	configuration, err := HandleConfigFile(configPath)
-	if err != nil {
-		l.Fatalf("error loading configuration: %s", err)
+	app.Commands = []cli.Command{
+		{
+			Name:  "server",
+			Usage: "Run the service as a server",
+			Flags: []cli.Flag{
+				cli.BoolFlag{
+					Name:  "automigrate",
+					Usage: "Run database migrations before starting.",
+				},
+			},
+
+			Action: cmdServer,
+		},
 	}
 
-	dbSession, err := NewMongoSession(configuration)
-	if err != nil {
-		l.Fatalf("failed to connect to DB: %v", err)
-	}
+	app.Action = cmdServer
+	app.Before = func(args *cli.Context) error {
 
-	err = MigrateDb(context.Background(), DbVersion, nil, dbSession)
-	if err != nil {
-		l.Fatalf("failed to migrate DB data: %v", err)
+		err := config.FromConfigFile(configPath, configDefaults)
+		if err != nil {
+			return cli.NewExitError(
+				fmt.Sprintf("error loading configuration: %s", err),
+				1)
+		}
+
+		// Enable setting config values by environment variables
+		config.Config.SetEnvPrefix("DEPLOYMENTS")
```

Comment:
`config.FromConfigFile()` already does that

Reply:
fixed